src/runtime/ProxyObject.cpp

```diff
@@ -512,6 +520,16 @@ Object::OwnPropertyKeyVector ProxyObject::ownPropertyKeys(ExecutionState& state)
     trapResultArray = Object::call(state, trap, handler, 1, arguments);
     auto trapResult = Object::createListFromArrayLike(state, trapResultArray, ((uint8_t)ElementTypes::String | (uint8_t)ElementTypes::Symbol));
 
+    // If trapResult contains any duplicate entries, throw a TypeError exception
+    size_t i = 1;
+    for (auto it = trapResult.begin(); it != trapResult.end(); ++it) {
+        for (auto it2 = trapResult.begin() + i; it2 != trapResult.end(); ++it2) {
+            if (it->equalsTo(state, *it2)) {
```

Comment:
`equalsTo` is for [strict equal comparison](https://www.ecma-international.org/ecma-262/10.0/index.html#sec-strict-equality-comparison). How about using `==` operator in `Value` instead of strict equal to check duplicated entries of trapResult?

Reply:
From the spec, we should check for the case of same string value, so `equalsTo` is necessary.
But how about iterating each value using the vector index rather than iterator here?
Following code for checking duplicated names in parser would be a case.
https://github.com/Samsung/escargot/blob/07a1e2b26ffec08b4a36ed3dc616a80ff085b7e8/src/parser/esprima_cpp/esprima.cpp#L463-L470
In the above case, it uses only two index variables and, IMHO it seems more intuitive.
What do you think about it?